TeamCode/src/main/java/org/firstinspires/ftc/teamcode/hardware/Drivebase.java

```diff
@@ -0,0 +1,275 @@
+     *
+     * @param inches    How many inches to strafe, negative is left.
+     * @param power     How fast to strafe. (0, 1].
+     * @param telemetry Pass this if you want the method to log to telemetry.
+     * @see Drivebase#driveForward
+     * @see Drivebase#turnAngle
+     * @see Drivebase#turnToAngle
+     */
+    public @Api void driveSideways(double inches, double power, @Nullable Telemetry telemetry) {
+        setMotorModes(DcMotor.RunMode.STOP_AND_RESET_ENCODER);
+
+        double target = inches * ENCODER_PER_INCH;
+
+        fldrive.setTargetPosition((int) target);
+        frdrive.setTargetPosition((int) -target);
+        bldrive.setTargetPosition((int) -target);
+        brdrive.setTargetPosition((int) target);
+
+        fldrive.setPower(power);
+        frdrive.setPower(-power);
+        bldrive.setPower(-power);
+        brdrive.setPower(power);
+
+        setMotorModes(DcMotor.RunMode.RUN_TO_POSITION);
+
+        waitForMotors(telemetry);
+
+        fldrive.setPower(0);
+        frdrive.setPower(0);
+        bldrive.setPower(0);
+        brdrive.setPower(0);
+    }
+
+    /**
+     * For auto. Turns to a specified angle, without resetting the heading.
+     *
+     * @param angle     How many degrees to turn. [-180, 180].
+     * @param power     How fast to turn. (0, 1].
+     * @param telemetry Pass this if you want this method to log to telemetry.
+     * @see Drivebase#driveSideways
+     * @see Drivebase#driveForward
+     * @see Drivebase#turnAngle
+     */
+    public @Api void turnToAngle(double angle, double power, @Nullable Telemetry telemetry) {
+        setMotorModes(DcMotor.RunMode.RUN_USING_ENCODER);
+
+        while (opModeIsActive.get() && Math.abs(getHeading() - angle) > 10) {
+            if (angle > 0) {
+                fldrive.setPower(power);
+                frdrive.setPower(-power);
+                bldrive.setPower(power);
+                brdrive.setPower(-power);
+            } else {
+                fldrive.setPower(-power);
+                frdrive.setPower(power);
+                bldrive.setPower(-power);
+                brdrive.setPower(power);
+            }
+
+            if (telemetry == null) continue;
+            addTelemetry(telemetry);
+        }
+
+        fldrive.setPower(0);
+        frdrive.setPower(0);
+        bldrive.setPower(0);
+        brdrive.setPower(0);
+    }
+
+    /**
+     * For auto. Turns a specified angle in degrees, as an offset from the current heading.
+     *
+     * @param angle     How many degrees to turn. [-180, 180].
+     * @param power     How fast to turn. (0, 1].
+     * @param telemetry Pass this if you want this method to log to telemetry.
+     * @see Drivebase#driveSideways
+     * @see Drivebase#driveForward
+     * @see Drivebase#turnToAngle
+     */
+    public @Api void turnAngle(double angle, double power, @Nullable Telemetry telemetry) {
+        imu.resetYaw();
+        turnToAngle(angle, power, telemetry);
```

Comment:
Do we need a turnAngle function in addition to the turnToAngle function? This seems like the initial turnToAngle function can be adjusted to be more appropriate to use, and eliminate the need for two, very similarly named functions.

Reply:
I think the naming could be improved, but I have quite often found myself needing to turn without resetting the heading. This will be fixed when we add PID turning